motionplan/tpSpaceRRT.go

```diff
@@ -788,51 +792,65 @@ func (mp *tpSpaceRRTMotionPlanner) setupTPSpaceOptions() {
 		goalMetricConstructor: defaultGoalMetricConstructor,
 	}
 
-	for _, curPtg := range mp.tpFrame.PTGSolvers() {
-		tpOpt.distOptions[curPtg] = mp.make2DTPSpaceDistanceOptions(curPtg)
+	mp.algOpts = tpOpt
+}
+
+func (mp *tpSpaceRRTMotionPlanner) ptgSolutionAndMetric(ptg tpspace.PTGSolver,
```

Comment:
Is there a reason to return the metric over the output of the metric?  It seems funny to be returning the metric over the value it produces.  There might be and I just I might not be following the code well enough

Reply:
This is doing the same thing the code we had before did. I just removed the duplication. As to why the code was like that, I defer to @biotinker

Reply:
We need both. When we check if a trajectory is valid (does not intersect obstacles), we allow that trajectory to partially extend if it hits an obstacle.

The metric is necessary so that we can compute the score of where the trajectory actually got to.